internal/transactionsubmission/transaction_worker.go

```diff
@@ -431,12 +432,16 @@ func (tw *TransactionWorker) buildAndSignTransaction(ctx context.Context, txJob
 
 	horizonAccount, err := tw.engine.HorizonClient.AccountDetail(horizonclient.AccountRequest{AccountID: txJob.ChannelAccount.PublicKey})
 	if err != nil {
-		err = fmt.Errorf("getting account detail: %w", err)
 		return nil, utils.NewHorizonErrorWrapper(err)
 	}
 
 	innerTx, err := tw.txHandler.BuildInnerTransaction(ctx, txJob, horizonAccount.Sequence, distributionAccount.Address)
 	if err != nil {
+		var hErr *utils.HorizonErrorWrapper
+		if errors.As(err, &hErr) {
+			return nil, hErr
+		}
```

Comment:
Why don't you just return a new horizon error wrapper?

```suggestion
		return nil, utils.NewHorizonErrorWrapper(err)
```